### PR TITLE
[WIP] [firebase_admob] Fix how views are displayed on iOS

### PR DESCRIPTION
**Pull request by bparrishMines**
## Description

Changes how the view is displayed on iOS by choosing the best layout guide depending on iOS version for 9.0+. On < 9.0, we set constraints manually. 

## Related Issues

Fixes https://github.com/flutter/flutter/issues/21634

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_admob/ios/Classes/FLTMobileAd.m [hunk 1]
```diff
@@ -101,70 +101,108 @@ - (NSString *)description {
...
+    [self.view.centerXAnchor constraintEqualToAnchor:layoutGuide.centerXAnchor
+                                            constant:_horizontalCenterOffset],
+    [self.view.leftAnchor constraintGreaterThanOrEqualToAnchor:layoutGuide.leftAnchor],
+    [self.view.rightAnchor constraintLessThanOrEqualToAnchor:layoutGuide.rightAnchor],
+  ]];
+}
+
+- (void)activateConstraintParentView:(UIView *_Nonnull)parentView
+                   verticalAttribute:(NSLayoutAttribute)verticalAttribute {
+  [parentView
```
Comment (blasten) [inline packages/firebase_admob/ios/Classes/FLTMobileAd.m:161]: The indentation is slightly different here than in the other ones. Is that intended?

File: packages/firebase_admob/ios/Classes/FLTMobileAd.m [hunk 2]
```diff
@@ -101,70 +101,108 @@ - (NSString *)description {
...
+  [parentView addConstraint:[NSLayoutConstraint constraintWithItem:self.view
+                                                         attribute:NSLayoutAttributeCenterX
+                                                         relatedBy:NSLayoutRelationEqual
+                                                            toItem:parentView
+                                                         attribute:NSLayoutAttributeCenterX
+                                                        multiplier:1
+                                                          constant:_horizontalCenterOffset]];
+
+  [parentView addConstraint:[NSLayoutConstraint constraintWithItem:self.view
+                                                         attribute:verticalAttribute
```
Comment (blasten) [inline packages/firebase_admob/ios/Classes/FLTMobileAd.m:185]: What about adding a comment that provides some context about the possible values that `verticalAttribute` can take?  Maybe add `NSAssert`.

File: packages/firebase_admob/example/ios/Podfile [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  system('rm -rf .symlinks')
+  system('mkdir -p .symlinks/plugins')
+  plugin_pods = parse_KV_file('../.flutter-plugins')
+  plugin_pods.each do |name, path|
+    symlink = File.join('.symlinks', 'plugins', name)
+    File.symlink(path, symlink)
+    pod name, :path => File.join(symlink, 'ios')
+  end
+
+  target 'firebase_admob_exampleUITests' do
```
Comment (bparrishMines) [inline packages/firebase_admob/example/ios/Podfile:77]: Everything in this file is generated except these 3 lines. They're required to run the tests in xcode.

File: packages/firebase_admob/example/ios/Podfile [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    pod 'OCMock','3.5'
+  end
```
Comment (bparrishMines) [inline packages/firebase_admob/example/ios/Podfile:79]: Everything in this file is generated except these 3 lines. They're required to run the tests in xcode.

File: packages/firebase_admob/example/ios/Runner.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -8,8 +8,10 @@
```
Comment (bparrishMines) [inline packages/firebase_admob/example/ios/Runner.xcodeproj/project.pbxproj:8]: A new test target was created, so the changes to this file need to be checked in.

File: packages/firebase_admob/ios/Classes/FLTMobileAd.m (hunk 1 above)
Comment (bparrishMines) [reply packages/firebase_admob/ios/Classes/FLTMobileAd.m:158]: This was done automatically by the clang formatter.

**Comment by bparrishMines** [comment]
@blasten I couldn't successfully add e2e tests, but I added a test that can be run locally. This test needs to run on 3 different iOS versions anyways which our CI doesn't currently do.

**Pull request closed without merging**